phpBB/phpbb/notification/method/board.php

```diff
@@ -0,0 +1,390 @@
+<?php
+/**
+*
+* This file is part of the phpBB Forum Software package.
+*
+* @copyright (c) phpBB Limited <https://www.phpbb.com>
+* @license GNU General Public License, version 2 (GPL-2.0)
+*
+* For full copyright and license information, please see
+* the docs/CREDITS.txt file.
+*
+*/
+
+namespace phpbb\notification\method;
+
+/**
+* In Board notification method class
+* This class handles in board notifications. This method is enabled by default.
+*
+* @package notifications
+*/
+class board extends \phpbb\notification\method\base
+{
+	/** @var string */
+	protected $notification_types_table;
+
+	/** @var string */
+	protected $notifications_table;
+
+	/**
+	* Notification Method Board Constructor
+	*
+	* @param \phpbb\user_loader $user_loader
+	* @param \phpbb\db\driver\driver_interface $db
+	* @param \phpbb\cache\driver\driver_interface $cache
+	* @param \phpbb\user $user
+	* @param \phpbb\auth\auth $auth
+	* @param \phpbb\config\config $config
+	* @param string $phpbb_root_path
+	* @param string $php_ext
+	* @param string $notification_types_table
+	* @param string $notifications_table
+	* @return \phpbb\notification\method\board
+	*/
+	public function __construct(\phpbb\user_loader $user_loader, \phpbb\db\driver\driver_interface $db, \phpbb\cache\driver\driver_interface $cache, $user, \phpbb\auth\auth $auth, \phpbb\config\config $config, $phpbb_root_path, $php_ext, $notification_types_table, $notifications_table)
+	{
+		$this->user_loader = $user_loader;
+		$this->db = $db;
+		$this->cache = $cache;
+		$this->user = $user;
+		$this->auth = $auth;
+		$this->config = $config;
+		$this->phpbb_root_path = $phpbb_root_path;
+		$this->php_ext = $php_ext;
```

Comment:
Actually I would prefer injecting all base dependencies to the base, and using inheritance in the config files